src/concurrent_list.hh

```diff
@@ -0,0 +1,80 @@
+/* This file is (c) 2021–2025 The GoldenDict-ng Community
+ * Part of GoldenDict. Licensed under GPLv3 or later, see the LICENSE file */
+
+#pragma once
+#include <list>
+using std::list;
+
+template< typename T >
+class concurrent_list
+{
+  QMutex mutex{};
+  std::list< T > list;
+
+public:
+
+  void push_back( const T & v )
+  {
+    QMutexLocker locker( &mutex );
+    list.push_back( v );
+  }
+
+  void remove( const T & v )
+  {
+    QMutexLocker locker( &mutex );
+    list.remove( v );
+  }
+
+  void clear()
+  {
+    QMutexLocker locker( &mutex );
+    list.clear();
+  }
+
+  bool empty()
+  {
+    QMutexLocker locker( &mutex );
+    return list.empty();
+  }
+  typename std::list< T >::iterator begin()
+  {
+    QMutexLocker locker( &mutex );
+    return list.begin();
+  }
```

Comment:
This is the accessor method.

The mutex only locks the accessing of the iterator, instead of the operation on the list.